tests/typecheck/models/test_contrib_models.yml

```diff
@@ -62,3 +62,45 @@
                   ...
               class MyUser(AbstractUser):
                   objects: ClassVar[MyUserManager] = MyUserManager()
+
+-   case: test_relation_specified_by_auth_user_model
+    main: |
+        from other.models import Other
+        reveal_type(Other().users.get())
+        reveal_type(Other.users.through)
+        reveal_type(Other.users.through().myuser)
+        reveal_type(Other.users.through.objects.get().myuser)
+
+        reveal_type(Other().user)
+
+        reveal_type(Other().unq_user)
+    out: |
+        main:2: note: Revealed type is "myapp.models.MyUser"
+        main:3: note: Revealed type is "Type[other.models.Other_users]"
+        main:4: note: Revealed type is "myapp.models.MyUser"
+        main:5: note: Revealed type is "myapp.models.MyUser"
+
+        main:7: note: Revealed type is "myapp.models.MyUser"
+
+        main:9: note: Revealed type is "myapp.models.MyUser"
+    custom_settings: |
+        INSTALLED_APPS = ('django.contrib.contenttypes', 'django.contrib.auth', 'myapp', 'other')
```

Comment:
What will happen if `django.contrib.auth` is not installed?

Reply:
You'd get an error asking you for explicit annotations/args of the field.

I can add a case for that too though

Reply:
Actually, not necessarily. It purely depends on if `settings.AUTH_USER_MODEL` can be resolve to a model or not, since we don't really touch the runtime..

Reply:
So it behaves the same as the runtime, since Django doesn't care if you put the lazy reference in a variable or not.